middleware/nethttp/testdata/example.wat

```diff
@@ -0,0 +1,36 @@
+;; This example module is written in WebAssembly Text Format to show the
+;; how a handler works and that it is decoupled from other ABI such as WASI.
+;; Most users will prefer a higher-level language such as C, Rust or TinyGo.
+(module $example
+    ;; http.log writes a message to the host console.
+    (import "http" "log" (func $http.log (param $ptr i32) (param $size i32)))
+
+    ;; http.next instructs the host to invoke the next handler.
+    (import "http" "next" (func $http.next))
```

Comment:
this is the key ABI change. the incoming function from the guest "handle" *may* invoke the downstream handler "next" or skip it. Because this is synchronous, we don't need to bother with returning a result like CONTINUE, etc. 

This dramatically simplifies the design as well, which is one main goal of this ABI.